### PR TITLE
Removal of SentinelInputStream

### PR DESCRIPTION
**Pull request by ctabin**
As per discussed [here](https://github.com/eclipse-ee4j/glassfish/pull/24369#issuecomment-1494719918), complete removal of `SentinelInputStream` and other related internal classes.

### CODE REVIEW

**Review by dmatej — APPROVED** [review]

**Comment by ctabin** [comment]
@dmatej Do you know how I could re-run the eclipsefdn/eca check ? I have completed the form several times but the job state doesn't change... Thanks !

**Comment by avpinchuk** [comment]
@ctabin, maybe try to to edit something in the description and save it again?

**Comment by ctabin** [comment]
Jenkins test please

**Comment by ctabin** [comment]
@dmatej Unfortunately, it doesn't help :-/ I'll amend my commit an push force...

**Comment by dmatej** [comment]
I will rerun that, I already tried, but it was stuck somehow yesterday or on Wednesday.

**Comment by dmatej** [comment]
Then we have to create an issue on gitlab...

**Comment by avpinchuk** [comment]
@dmatej, on Wed build works with some ant tests failed randomly by timeout after 1h. Yesterday build process wait 6h on scanning project phase and failed after 6h timeout...

**Comment by dmatej** [comment]
> @dmatej, on Wed build works with some ant tests failed randomly by timeout after 1h. Yesterday build process wait on scanning project phase and failed after 6h timeout...

There were some issues since friday, on tuesday it worked again, but I forgot to ask about the ECA issue.
Btw yesterday something broke again ... https://www.eclipsestatus.io/

Seems no related issue to ECA at this moment ... https://www.eclipsestatus.io/ ... I will create one.

**Comment by dmatej** [comment]
And yet this one: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/2964

File: appserver/connectors/connectors-internal-api/src/main/java/com/sun/appserv/connectors/internal/api/ConnectorsClassLoaderUtil.java [hunk 1]
```diff
@@ -118,7 +119,11 @@ private ConnectorClassFinder createRARClassLoader(final ClassLoader parent, Stri
                 if (processEnv.getProcessType().isEmbedded()) {
                     events.register(event -> {
                         if (event.is(EventTypes.PREPARE_SHUTDOWN)) {
-                            ccf.done();
+                            try {
+                                ccf.close();
+                            } catch(IOException ioe) {
+                                _logger.log(Level.WARNING, "error.closing.connector.classloader", ioe);
```
Comment (dmatej) [inline appserver/connectors/connectors-internal-api/src/main/java/com/sun/appserv/connectors/internal/api/ConnectorsClassLoaderUtil.java:125]: I think this key doesn't exist anywhere ... you can just write an english message what failed, ie. Closing of the Connector Class Finder failed.

**Pull request merged**